api/dimensions.go

```diff
@@ -175,10 +159,17 @@ func (api *DatasetAPI) getDimensionOptions(w http.ResponseWriter, r *http.Reques
 
 	b, err := func() ([]byte, error) {
 
-		// get limit and offset from query parameters, or default values
-		limit, offset, err := api.getLimitAndOffset(r.URL.Query())
+		// get limit from query parameters, or default value
+		limit, err := getPositiveIntQueryParameter(r.URL.Query(), "limit", api.defaultLimit)
+		if err != nil {
+			log.Event(ctx, "failed to obtain limit from request query paramters", log.ERROR, log.Error(err), logData)
```

Comment:
Would it be better to stop doing log.Error(err) that puts a stack dump into the log file for Error's that are not program errors ... BUT are in fact a user mistake/error.
Consider replacing line 165 with:
			logData["user error"] = fmt.Sprintf("%+v", err)
			log.Event(ctx, "failed to obtain limit from request query paramters", log.ERROR, logData)

where the log.Error(err) is removed and its err is put into logData[] and clearly identified as a 'user error' (not a program error).

And similarly for any other added log.Error()'s in this feature branch.

Reply:
I generalised this to `handleDimensionsErr`, which only logs the stack trace for InternalServer errors, and adds `user_error` to logData otherwise (BadRequest, NotFound, etc.)

Reply:
looking at code ... see questions in slack